### ISSUE TITLE
Test ICD: vkMapMemory should return a pointer aligned to the strictest alignment mentioned in the spec

### ISSUE THREAD

**Issue by arno-lunarg** [KhronosGroup/Vulkan-ValidationLayers#8776]
It currently does not do that, leading to unexpected alignment issues when filling shader binding tables in the ray tracing framework

**Comment by ziga-lunarg** [comment]
Should this be a CTS test?

**Comment by spencer-lunarg** [comment]
@ziga-lunarg no, this is an issue with our TestICD (the new Mock ICD) ... we have no plans to make a CTS conformance mock ICD and so a CTS test would not help

**Comment by spencer-lunarg** [comment]
@arno-lunarg what does 

> aligned to the strictest alignment mentioned in the spec

mean, how would TestICD determine this, another issue is things like `shaderGroupBaseAlignment` we set in the profile JSON and not the ICD itself

**Comment by arno-lunarg** [comment]
> @arno-lunarg what does
> 
> > aligned to the strictest alignment mentioned in the spec
> 
> mean, how would TestICD determine this, another issue is things like `shaderGroupBaseAlignment` we set in the profile JSON and not the ICD itself

No way to query that directly, so I would make sure the returned pointer is aligned to 8192, that should be enough

**Comment by spencer-lunarg** [comment]
what did `8192` come from?

**Comment by arno-lunarg** [comment]
Totally made up, just go for a somewhat big power of 2!
